app.rb

```diff
@@ -0,0 +1,54 @@
+require 'sinatra'
+require 'sinatra/reloader'
+require 'active_record'
+
+
+require_relative 'db/connection'
+
+require_relative 'models/pokemon'
+
+get "/pokemons" do
+  @pokemons = Pokemon.all
+  erb :"pokemon/index"
+end
+
+get "/pokemons/:img_url" do
+  @pokemon = Pokemon.find(params[:img_url])
+  erb :"pokemon/index"
+end
+
+post "/pokemons" do
+  @pokemon = Pokemon.create(name: params[:name], poke_type: params[:poke_type], cp: params[:cp], img_url: params[:img_url])
+  redirect "pokemons/#{@pokemon.id}"
+end
+
+get "/pokemons/:id" do
+  @pokemon = Pokemon.find(params[:id])
+  erb :"pokemon/show"
+end
+
```

Comment:
you only need a method that finds a Pokemon by id, but not the other attributes `cp` `name` etc